### ISSUE TITLE
Magic Mouse horizontal scrolling does not work, but Macbook trackpad side scroll works as expected

### ISSUE THREAD

**Issue by s-daveb** [debauchee/barrier#1501]
### What happened?

# Configuration
```
Macbook Pro x86_64 2017, macOS 12.1, Barrier 2.40-release-433126e7 
Macbook Air  M1    2020, macOS 12.1, Barrier 2.40-release-3e0d758b (x86_64)
```

## Summary 
I'm unsure if this is related to #591,...but it's been my experience that when I use my magic mouse on my Macbook Pro, I'm unable to use it to scroll a browser window horizontally on my Macbook Air Client.  The reverse however, works as expected - using the Macbook Air as a server, and the Macbook Pro as a client, I can scroll in any direction using the macbook Air trackpad. 

I'm not sure if this would count as a gesture and should be part of #591, or if trackpad side scrolling as added after #591 was created, nor am I sure  that horizontal scrolling is detected on a magic mouse works similarly to a trackpad, but I thought this should maybe be a separate issue. 

### Version

v2.4.0

### Git commit hash (if applicable)

433126e7

### If applicable, where did you install Barrier from?

Homebrew

### What OSes are you seeing the problem on? (Check all that apply)

macOS

### What OS versions are you using?

macOS 12.1

**Comment by ElhemEnohpi** [comment]
I think you're correct that this is a separate issue. It's not related to #591 because "multitouch gestures" are the things unique to Macs - three- and four-finger gestures, squeeze and pinch, etc., which are not supported at all in Barrier. Horizontal scrolling is a universal, cross-platform feature that's part of the basic mouse scrolling functionality of Barrier. If it doesn't work with a Magic Mouse, I'm not sure why not - I don't have one to test, sorry - but it's a different issue than #591.

Btw., you say it "works as expected" when using the trackpad. Do you not find that the horizontal scroll direction is reversed, as described in #476?

**Comment by s-daveb** [comment]
Ah, I think I understand why it doesn't work now… thanks for taking the time to explain. 
And actually as far as #476, I'm not seeing that reversed behavior on my Macbook Air Touchpad. It works as expected.

**Comment by bgbruno** [comment]
@ElhemEnohpi @s-daveb I think this issue is related to "OS Setting" and #barrier somehow read "Default PC behaviour" instead of
```
System Preferences > Mouse > Scroll direction: Natural
```
===
```
System Preferences > Trackpad > Scroll & Zoom > Scroll direction: Natural
```

**Comment by Chiwency** [comment]
> Ah, I think I understand why it doesn't work now… thanks for taking the time to explain. And actually as far as #476, I'm not seeing that reversed behavior on my Macbook Air Touchpad. It works as expected.

I have the same problem, how did you fix it?

**Comment by s-daveb** [comment]
> > Ah, I think I understand why it doesn't work now… thanks for taking the time to explain. And actually as far as #476, I'm not seeing that reversed behavior on my Macbook Air Touchpad. It works as expected.
> 
> I have the same problem, how did you fix it?

I honestly do not remember... 
I switched to a fork of Barrier that seems to be more maintained. 
Check out https://github.com/input-leap/input-leap

**Comment by Chiwency** [comment]
Thank you, It helps a lot, and I will switch it too.  What's more, I had found the true reason. It's the X-Mouse Button Control disabled my mouse scroll.
